db/friend.go

```diff
@@ -1,5 +1,84 @@
 package db
 
+import (
+	"fmt"
+	"log"
+)
+
 type Friend struct {
-	MAC, IP string
+	DisplayName, MAC, IP string
+}
+
+// Checks if the two users are friends given friend's display name
+func AreFriendsName(username string, friendDisplayName string) bool {
+	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
+	friends := ExecuteFriendsQuery(query)
+	return len(friends) > 0
+}
+
+// Checks if the two users are friends given friend's MAC Address
+func AreFriendsMac(username string, friendMacAddress string) bool {
+	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_mac_address=\"%s\"", friendsTableName, username, friendMacAddress)
+	friends := ExecuteFriendsQuery(query)
+	return len(friends) > 0
+}
+
+// Get friend from database
+func GetFriendByDisplayName(friendDisplayName string) *Friend {
```

Comment:
Same thing here, this needs to be have the username of the user as another argument if you are querying by friendDisplayName.